pyuvsim/telescope.py

```diff
@@ -37,3 +41,278 @@ def __eq__(self, other):
+        except ValueError as err:
+            if self.string_mode:
+                self._str_beam_list.pop()
+            else:
+                self._obj_beam_list.pop()
+            raise err
+
+    def _str_to_obj(self, beam_model):
+        # Convert beam strings to objects.
+        if isinstance(beam_model, (AnalyticBeam, UVBeam)):
+            return beam_model
+        if beam_model.startswith('analytic'):
+            bspl = beam_model.split('_')
+            model = bspl[1]
+
+            to_set = {}
+            for extra in bspl[2:]:
+                par, val = extra.split('=')
+                full = self._float_params[par]
+                to_set[full] = float(val)
+
+            return AnalyticBeam(model, **to_set)
+
+        path = beam_model  # beam_model = path to beamfits
+        uvb = UVBeam()
+
+        uvb.read_beamfits(path)
+        for key, val in self.uvb_params.items():
+            setattr(uvb, key, val)
+        uvb.extra_keywords['beam_path'] = path
+        return uvb
+
+    def _obj_to_str(self, beam_model):
+        # Convert beam objects to strings that may generate them.
+
+        if isinstance(beam_model, str):
+            return beam_model
+        if isinstance(beam_model, AnalyticBeam):
+            btype = beam_model.type
+
+            bm_str = 'analytic_' + btype
+            for abbrv, full in self._float_params.items():
+                val = getattr(beam_model, full)
+                if val is not None:
+                    bm_str += '_' + abbrv + '=' + str(val)
+            return bm_str
+
+        # If not AnalyticBeam, it's UVBeam.
+        try:
+            path = beam_model.extra_keywords['beam_path']
+        except KeyError:
+            raise ValueError("Need to set 'beam_path' key in extra_keywords for UVBeam objects.")
+
+        return path
+
+    def set_str_mode(self):
+        """
+        Set string_mode = True, converting object beams to their
+        string representations and putting UVBeam parameters into the
+        uvb_params dictionary.
+
+        Raises
+        ------
+        ValueError:
+            If UVBeam objects in the list have inconsistent UVBeam parameters.
+        ValueError:
+            If any UVBeam objects lack a "beam_path" keyword in the "extra_keywords"
+            attribute, specifying the path to the beamfits file that generates it.
+        """
+        if not self._obj_beam_list == []:
+            # Convert object beams to string definitions
+            new_uvb_par = self._scrape_uvb_params(self._obj_beam_list, strict=True)
+            self.uvb_params.update(new_uvb_par)
+            self._str_beam_list = [self._obj_to_str(bobj) for bobj in self._obj_beam_list]
+        self._obj_beam_list = []
+        self.string_mode = True
+
+    def set_obj_mode(self):
+        """
+        Set string_mode = False, initializing beam objects from the strings.
+        Additional parameters on UVBeam objects will be set from the uvb_params dictionary.
```

Comment:
This docstring will error with flake8 when we turn on docstring testing in Steven's PR -- The first line should be a single line sentence (ending in a period). More details should follow in text after a blank line.